### PR TITLE
Fix/validation message

### PR DESCRIPTION
**Pull request by andreaDiClaudio**
Fixed Validation message on signup and login. 

/*BACKEND*/
Logic:
Error Message Handling in the Form:
The SignupForm component receives an errorMessages prop, which is an array of error messages returned from the
backend.

The getFieldErrorMessage utility function is used to find the relevant error message for each field based on the field name.
Instead of matching the entire error message, the second parameter in the getFieldErrorMessage function is used to match a part of the error message that corresponds to a specific field.
For example, when validating the email field, the second parameter "Email" is used to match the error message that contains "Email must be a valid email address".
This allows the form to display the correct validation message for each field. If no error for that field is found, the validityMsg will be undefined, and the input field is considered valid.

Field Validation:
Each input field (TextInput) checks for an error related to its specific field (e.g., "First Name", "Last Name", "Email", etc.) using the getFieldErrorMessage function.
The function performs a case-insensitive search to find an error message that contains a specific field's name or part of the message.

/*Frontend*/
Now also the frontend has the input validation and the textInput has the 'required' attribute field. Data gets trimmed and validated before being sent to backend (where will it be validated again)

### CODE REVIEW

File: client/src/routes/accounts.lazy.tsx
Comment (amorea9) [inline client/src/routes/accounts.lazy.tsx]: There's a small discrepancy here as in the backend the min lenght is set as 6 characters and in the frontend at 8. We might want to adjust that to be the same so we don't get inconsistent data in the future

File: client/src/routes/accounts.lazy.tsx
Comment (amorea9) [inline client/src/routes/accounts.lazy.tsx]: To prevent fields from being empty you can just add a check for it in this function. To make it even better you can add the required attribute to the textInput fields that need it

**Review by amorea9 — APPROVED** [review]

**Comment by andreaDiClaudio** [comment]
Waiting for fix/user-fetch to be merged into main before resolving conflicts and merging

**Review by amorea9 — APPROVED** [review]

**Pull request merged**
